hal/src/rtl872x/core_hal.c

```diff
@@ -596,6 +596,11 @@ static void Init_Last_Reset_Info()
         // Clear backup registers
         HAL_Core_Write_Backup_Register(BKP_DR_02, 0);
         HAL_Core_Write_Backup_Register(BKP_DR_03, 0);
+
+        // FIXME: System.reset() is currently using watchdog to reset device.
```

Comment:
This needs to be fixed by perhaps setting some flag in https://github.com/particle-iot/device-os/blob/develop/hal/src/rtl872x/rtl_sdk_support.cpp#L191

Also, does that reset function still work when WDG is enabled?

Reply:
yes, the reset function is working, cos the watchdog is reconfigurable.